pkg/gadgets/profile/cpu/tracer/tracer.go

```diff
@@ -327,30 +327,13 @@ func (t *Tracer) install() error {
 		return fmt.Errorf("failed to load ebpf program: %w", err)
 	}
 
-	mapReplacements := map[string]*ebpf.Map{}
-	filterByMntNs := false
-
-	if t.config.MountnsMap != nil {
-		filterByMntNs = true
-		mapReplacements[gadgets.MntNsFilterMapName] = t.config.MountnsMap
-	}
-
 	consts := map[string]interface{}{
-		"kernel_stacks_only":      t.config.KernelStackOnly,
-		"user_stacks_only":        t.config.UserStackOnly,
-		gadgets.FilterByMntNsName: filterByMntNs,
-	}
-
-	if err := spec.RewriteConstants(consts); err != nil {
-		return fmt.Errorf("error RewriteConstants: %w", err)
-	}
-
-	opts := ebpf.CollectionOptions{
-		MapReplacements: mapReplacements,
+		"kernel_stacks_only": t.config.KernelStackOnly,
+		"user_stacks_only":   t.config.UserStackOnly,
 	}
 
-	if err := spec.LoadAndAssign(&t.objs, &opts); err != nil {
-		return fmt.Errorf("failed to load ebpf program: %w", err)
+	if err := gadgets.LoadeBPFSpec(t.config.MountnsMap, spec, consts, &t.objs); err != nil {
+		return fmt.Errorf("loading ebpf program: %w", err)
```

Comment:
nit: program -> spec. (It could happen that the error is related to a map, so program will be misleading in that case)

Reply:
✅  Perfect. I wasn't 100% convinced that the error message is the right one.